### ISSUE TITLE
Making the copyparty interface more accessible for screenreader users

### ISSUE THREAD

**Issue by domasofan** [9001/copyparty#1165]
Hi all,

I just found out about copyparty and i  like the concept and how it works.
As a blind user i am using a screenreader to use my computers and my phone.
Unfortunately it is pretty difficult to use the Web UI because the links and buttons are just labeled with emojis.
So the screenreader just reads which emoji it is if it knows it.
It would be great if the UI would be more accessible with labels maybe just the screenreader can see.
As far as i know this is available in HTML 5 or WAI ARIA.
Maybe someone could have a look into this.

It would be great if we can make this party more accessible and inclusive for everyone. 🥳🎉🪩🧑‍🦽🧑‍🦯

Greetings from Austria,
Simon

**Comment by 9001** [comment]
I will try to look into this before the next release; given that the following approach will work, I plan to insert aria attributes into the various elements on the page from javascript, to avoid increasing the size of the template html files. This should also make localization possible, and not have everything in english. The first step will be to learn what tools are available to verify what I'm doing, since this is a subject I've never looked into before -- but I expect there is enough documentation that I should be able to figure this out on my own. Thanks for the nudge!

**Comment by Anubarak** [comment]
@9001 web developer here with certain accessibility certificates: in my country we are forced to make every webshop fully accessible and from my experience in the last 6 years: a little accessibility is far better than none of course, but even a simple website takes several days to make it accessible.

Furthermore not "bloating" the HTML directly is sometimes a bad idea. Some attributes can't be added during runtime and must be present during page load. (however most can)

I would expect a higher level of accessibility for your project might take about 150 hours as many required things needs to be rewritten entirely. From my experience most developers - including me when I started  - won't even consider most of the things that needs to be done.

However adding a few aria attributes is a great first step but even then it will take MUCH time to figure out the correct one to make it good. (except a title attribute).

I am not trying to discourage you - I just want to make you aware of the insane time investment It takes.

**Comment by domasofan** [comment]
Hi all,

As far as i have seen we might be already pretty good when we do the following:
* label all links, buttons, form fields, etc.
* In the file list add checkboxes to select the files.
* also maybe when there are clickable elements where you can toggle something (on/off) it might be interesting to use togglebuttons or checkboxes.

The WCAG and the ARIA specs are nice.
Though sometimes you can get good accessibility without them. 😉

If we have a GUI toolkit or some widgets which are made externally we might have to open some issues in their repos.
So it might be fixed by them which helps others too.


If someone wants to test things first before committing to the repo you can also send me an email with the html and java script files to my address: simon [dot] eigeldinger [at] vol [dot] at

Greetings,
Simon
